pytato/array.py

```diff
@@ -585,30 +585,33 @@ def _binary_op(self,
             other: Union[Array, Number],
             get_result_type: Callable[[DtypeOrScalar, DtypeOrScalar], np.dtype[Any]] = np.result_type,  # noqa
             reverse: bool = False) -> Array:
+        import pytato.utils as utils
 
-        def add_indices(val: prim.Expression) -> prim.Expression:
-            if self.ndim == 0:
+        def add_indices(val: prim.Expression,
+                        s: ShapeType, r: ShapeType) -> prim.Expression:
```

Comment:
```suggestion
                        shape: ShapeType, result_shape: ShapeType) -> prim.Expression:
```
(not wild about `s` and `r` as argument names)